core/textile/client.go

```diff
@@ -198,35 +236,9 @@ func getHubThreadsClient(host string) *threadsClient.Client {
 	return tc
 }
 
-// StartAndBootstrap starts a Textile Client and also initializes default resources for it like a key pair and default bucket.
-func (tc *textileClient) Start(ctx context.Context, cfg config.Config) error {
-	// Create key pair if not present
-	log.Debug("Starting Textile Client: Getting key status...")
-	_, err := tc.kc.GetStoredPublicKey()
+func (tc *textileClient) initialize(ctx context.Context) error {
+	buckets, err := tc.listBuckets(ctx)
```

Comment:
@dmerrill6 

I might be wrong but my understanding is that `listBuckets()` return `nil, err` `if tc.isRunning == false || tc.isInitialized == false` and this might be exactly the case here `since tc.isInitialized = true` is only done later.

Reply:
I updated `ListBuckets` to do the `requiresRunning` check, but `initialize` is calling `listBuckets` instead, which doesn't check for these conditions.

The idea behind this is that if `ListBuckets` (the public method) is called from outside and the client is not running, we return an error.

Reply:
@dmerrill6 

You are right! My bad!